Example/Example/SeriesDetailViewController.swift

```diff
@@ -9,6 +9,92 @@
 import Foundation
 import BothamUI
 
-class SeriesDetailViewController : ExampleViewController {
+private struct Config {
+    static let numberOfColumns = 3
+    static let cellHeight = 174
+    static let headerHeight = 534
+    static let footerHeight = 20
+    static let cellMargin = 20
+}
+class SeriesDetailViewController : ExampleViewController, BothamCollectionViewController, SeriesDetailUI, UICollectionViewDelegateFlowLayout, UICollectionViewDelegate, UICollectionViewDataSource {
+
+
+    var dataSource: BothamCollectionViewDataSource<Comic, ComicCollectionViewCell>!
+    var header: Series?
+
+    @IBOutlet weak var collectionView: UICollectionView!
+
+    override func viewDidLoad() {
+        configureNavigationBar()
+        configureCollectionView()
+        super.viewDidLoad()
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        navigationController?.navigationBar.setBackgroundImage(nil, forBarMetrics: UIBarMetrics.Default)
+        navigationController?.navigationBar.backgroundColor = UIColor.navigationBarColor
+        navigationController?.navigationBar.shadowImage = nil
+        navigationController?.navigationBar.translucent = false
+    }
+
+    func collectionView(
```

Comment:
The problem I had was with the header. To be able to add a header I needed to implement the DataSource protocol and that's why I had an instance of BothamDataSource and at the same time the view controller was implementing that protocol, to add the header behaviour and at the same time delegate the rest of the protocol calls to the BothamDataSource.

I've extracted the header code to a SeriesDetailHeaderDataSource extending from BothamDataSource to remove this protocols from the view controller.